Sources/FoundationNetworking/URLSession/HTTP/HTTPURLProtocol.swift

```diff
@@ -241,11 +241,35 @@ internal class _HTTPURLProtocol: _NativeProtocol {
     
     static let dateFormatter: DateFormatter = {
         let x = DateFormatter()
-        x.locale = NSLocale.system
-        x.dateFormat = "EEE',' dd' 'MMM' 'yyyy HH':'mm':'ss zzz"
         return x
     }()
     
+    static func date(from string: String) -> Date? {
+        // https://tools.ietf.org/html/rfc2616#section-3.3.1
+        // HTTP applications have historically allowed three different formats
+        // for the representation of date/time stamps
+        
+        // RCF 822 --- Sun, 06 Nov 1994 08:49:37 GMT
+        self.dateFormatter.dateFormat = "EEE, dd MMM yyyy HH:mm:ss zzz"
```

Comment:
There are costs associated with changing the settings of a formatter, and I’d rather just have three different formatters ready here.

Reply:
```swift
    private var _dateFormat: String? { willSet { _reset() } }
    open var dateFormat: String! {
        get {
            guard let format = _dateFormat else {
                return CFDateFormatterGetFormat(_cfObject)._swiftObject
            }
            return format
        }
        set {
            _dateFormat = newValue
        }
    }
```

looks like anything was cost

Reply:
Sorry, I don’t understand your comment.

I was referring to the resources that CFDateFormatterSet… will destroy and recreate every time it’s called. I’d rather just have three different date formatters so we can amortize that call.

Reply:
(That call is in _reset().)

Reply:
OK，I got it